### ISSUE TITLE
Port ResNet50 to the new model architecture style

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#78]

**Comment by MrinalTyagi** [comment]
Hi. I would like to work on this issue if it's open for contributions.

**Comment by mradul2** [comment]
Hey, I would like to work on this issue, what exactly do you mean by porting into the **new model architecture style** ?

**Comment by bhack** [comment]
@qlzh727 A we have already discussed about this but I don't know if we are ready:

- Training scripts to reproduce the results
- [TFDS integration](https://github.com/tensorflow/datasets) when available instead of populating the repo of custom scripts for dataset adaptation/loading.

**Comment by bhack** [comment]
Yes I hope that we are not going to accumulate too much single shared scripts that in the end over the time are quite complex and always under a combinatorial explosion risk.

I suppose that some json/yaml/etc metadata overlay/override design could be required to customize/specialize training jobs.

**Comment by qlzh727** [comment]
True. I think we will need some design for the overall input pipeline and training logic for this case, so that it will configurable, composable and sharable.

**Comment by bhack** [comment]
@qlzh727 At at the end one day it could be really nice if, with an approval label, we could schedule the training job from CI directly on GKE (sponsored) internal resources.

**Comment by MrinalTyagi** [comment]
@qlzh727 I am able to understand all the points you highlighted in the new model style and would like to proceed with it if you allow.

**Comment by AdityaKane2001** [comment]
@qlzh727 

I believe if the central requirement is reproducibility, there can be a one-to-one mapping between a data input pipeline and model. For example:

```py
# dataloaders.py
class ImageNetDataset:
    # Can contain utilites to read and parse TFRecords and other mandatory things
    def __init__(self, model_agnostic_params=their_default_values): # like num_classes, image_channels, etc. 
        ...
    def make_dataset(self):
        # This method should be implemented by inherited classes, attending to particular needs of their models
        raise NotImplementedError

class EfficientNetDataset(ImageNetDataset):
    def __init__(self, model_specific_params=their_default_values):
        ...

# models.py
def EfficientNet(model_specific_params):
    ...
    efficientnet = tf.keras.Model(inputs=inputs, outputs=x, name="efficientnet_b0")
    return efficientnet

# train.py
# We take the model name to be trained from the user and train the model with all the default, reproducible hyperparameters
...
if model_str == "efficientnet":
    train_ds, val_ds = EfficientNetDataset().make_dataset()
    model = EfficientNet()

with strategy.scope():
    history = model.fit(
    train_ds,
   	epochs=train_cfg.total_epochs,
   	validation_data=val_ds,
   	callbacks=callbacks,
)
```
This is just me thinking out loud. Please correct me if I'm wrong.

**Comment by LukeWood** [comment]
Sorry all - this is not open for contributions as there is some nuance as to how this will be done.  Hope everyone understands!
